test/milton/MiltonEvents.t.sol

```diff
@@ -0,0 +1,525 @@
+            AmmTypes.OpenSwapMoney({
+                totalAmount: TestConstants.USD_10_000_18DEC, // totalAmount
+                collateral: TestConstants.TC_COLLATERAL_18DEC, // collateral
+                notional: 99670098970308907327800, // notional
+                openingFeeLPAmount: TestConstants.TC_OPENING_FEE_18DEC, // openingFeeLPAmount
+                openingFeeTreasuryAmount: TestConstants.ZERO, // openingFeeTreasuryAmount
+                iporPublicationFee: TestConstants.TC_IPOR_PUBLICATION_AMOUNT_18DEC, // iporPublicationFee
+                liquidationDepositAmount: TestConstants.TC_LIQUIDATION_DEPOSIT_AMOUNT_18DEC // liquidationDepositAmount
+            }), // money
+            block.timestamp, // openTimestamp
+            block.timestamp + TestConstants.SWAP_DEFAULT_PERIOD_IN_SECONDS, // endTimestamp, 28 days, PERIOD_28_DAYS_IN_SECONDS
+            MiltonTypes.IporSwapIndicator({
+                iporIndexValue: TestConstants.PERCENTAGE_3_18DEC, // iporIndexValue
+                ibtPrice: 1 * TestConstants.D18, // ibtPrice
+                ibtQuantity: 99670098970308907327800, // ibtQuantity
+                fixedInterestRate: TestConstants.PERCENTAGE_4_18DEC // fixedInterestRate, 4%
+            }) // indicator
+        );
+        miltonDai.itfOpenSwapPayFixed(
+            block.timestamp, // openTimestamp
+            TestConstants.USD_10_000_18DEC, // totalAmount
+            TestConstants.PERCENTAGE_6_18DEC, // acceptableFixedInterestRate, 6%
+            TestConstants.LEVERAGE_18DEC // leverage, LEVERAGE_18DEC
+        );
+    }
+
+    function testShouldEmitEventWhenOpenReceiveFixedSwap18Decimals() public {
+        // given
+        MockTestnetToken daiMockedToken = getTokenDai();
+        ItfIporOracle iporOracle = getIporOracleAsset(_userOne, address(daiMockedToken), 0);
+        IpToken ipTokenDai = getIpTokenDai(address(daiMockedToken));
+        MockCase0Stanley stanleyDai = getMockCase0Stanley(address(daiMockedToken));
+        MiltonStorage miltonStorageDai = getMiltonStorage();
+        MockCase0MiltonDai miltonDai = getMockCase0MiltonDai(
+            address(daiMockedToken),
+            address(iporOracle),
+            address(miltonStorageDai),
+            address(_miltonSpreadModel),
+            address(stanleyDai)
+        );
+        MockCase0JosephDai josephDai = getMockCase0JosephDai(
+            address(daiMockedToken),
+            address(ipTokenDai),
+            address(miltonDai),
+            address(miltonStorageDai),
+            address(stanleyDai)
+        );
+        prepareApproveForUsersDai(_users, daiMockedToken, address(josephDai), address(miltonDai));
+        prepareMilton(miltonDai, address(josephDai), address(stanleyDai));
+        prepareJoseph(josephDai);
+        prepareIpToken(ipTokenDai, address(josephDai));
+        // when
+        _miltonSpreadModel.setCalculateQuoteReceiveFixed(TestConstants.PERCENTAGE_2_18DEC); // 2%
```

Comment:
If you are doing copy paste in tests please do it in the same order "setCalculateQuotePayFixed" was at the beginning.